app/src/main/java/edu/usf/sas/pal/muser/firebase/models/CurrentSongInfo.kt

```diff
@@ -0,0 +1,32 @@
+package edu.usf.sas.pal.muser.firebase.models
```

Comment:
I think that we should have another class, maybe `SongAction`, that contains the action we're trying to capture (e.g., play, pause, skip) and a timestamp that the action occurred and a nested `SongData` object (the song that the action was applied to). See https://github.com/CUTR-at-USF/MUSER/issues/17#issuecomment-658419090 for a partial list of actions we've previously thought of. Could you take a shot at that and add to this PR?

Alternately, we include include the action information as part of this model object, but it seems like a better design to me to keep them separate. Let me know your thoughts.

Reply:
Great idea. Alternatively, we can also make the `SongAction` class nested with the `SongData` class. When the user opens a song (plays it from the beginning?), we could have an additional action saying "Opened" or we could just stick to "Play". Ultimately, when a user interacts with the track, we'll have information about the song and the action that was performed on it. Please let me know your opinions.

Reply:
Given that we're really storing a stream of actions in Firebase, I think it's better for the action to be the parent entity.

Also, for timestamps, we'll want to capture both https://developer.android.com/reference/java/lang/System#currentTimeMillis() as well as https://developer.android.com/reference/java/lang/System#nanoTime() for each event. We did this for the activity behavior tracking app, because nanoTimes are logged by lower level hardware and used to measure elapsed time, but you need currentTimeMillis to resolve them to a wall clock time.

Reply:
I can understand that. Thank you. and also, thank you for the links.